sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -1170,7 +1311,12 @@ object SparkSession extends Logging {
    * @since 2.0.0
    */
   def setActiveSession(session: SparkSession): Unit = {
+    clearActiveSession()
     activeThreadSession.set(session)
+    if (session != null) {
+      session.sparkContext.setLocalProperty(SPARK_SESSION_UUID_PROPERTY_KEY, session.sessionUUID)
+      session.userDefinedToRealTagsMap.values().asScala.foreach(session.sparkContext.addJobTag)
```

Comment:
I am not sure if this is bullet proof. It is better to set this up in `SQLExecution`.

Reply:
Done!